File: java/1785.构成特定和需要添加的最少元素.java
import java.util.LinkedList;
import java.util.List;

/*
 * @lc app=leetcode.cn id=1785 lang=java
 * @lcpr version=21103
 *
 * [1785] 构成特定和需要添加的最少元素
 */

// @lc code=start
class Solution {
    public int minElements(int[] nums, int limit, int goal) {
        long res = 0;
        for (int i : nums) {
            res+=i;
        }
        long gap = Math.abs(res-goal);

        return (int)((gap+limit-1)/limit); 

    }
}
// @lc code=end



/*
// @lcpr case=start
// [1,-1,1]\n3\n-4\n
// @lcpr case=end

// @lcpr case=start
// [1,-10,9,1]\n100\n0\n
// @lcpr case=end

 */

File: cpp/1785.构成特定和需要添加的最少元素.cpp
/*
 * @lc app=leetcode.cn id=1785 lang=cpp
 * @lcpr version=21102
 *
 * [1785] 构成特定和需要添加的最少元素
 */
#include <algorithm>
#include <array>
#include <bitset>
#include <climits>
#include <cmath>
#include <cstdlib>
#include <deque>
#include <functional>
#include <iostream>
#include <list>
#include <queue>
#include <stack>
#include <tuple>
#include <unordered_map>
#include <unordered_set>
#include <utility>
#include <vector>
using namespace std;
// @lc code=start
class Solution {
public:
    int minElements(vector<int>& nums, int limit, int goal) {
        long long sum=0;
        for (auto i : nums) {
            sum+=i;
        }
        long long diff = abs(sum - goal);
        return (diff + limit - 1) / limit;
    }
};
// @lc code=end



/*
// @lcpr case=start
// [1,-1,1]\n3\n-4\n
// @lcpr case=end

// @lcpr case=start
// [1,-10,9,1]\n100\n0\n
// @lcpr case=end

 */


